cpp/src/IceGrid/AdminI.cpp

```diff
@@ -129,7 +129,7 @@ namespace
 
     private:
         string _id;
-        ServerPrxPtr _proxy;
+        optional<ServerPrx> _proxy;
```

Comment:
I don't see how _proxy can be null. It's just convenience for the constructor?

Reply:
We can fix this by using a helper function, and moving the `SynchronizationException` handling to the helper method.

Reply:
I will try to fix in a follow up PR.